src/towncrier/test/test_create.py

```diff
@@ -3,7 +3,9 @@
 
 import os
 from textwrap import dedent
+
 from twisted.trial.unittest import TestCase
+import mock
```

Comment:
I don't like using mock but Click does not yet provide a means for testing code that uses `click.edit()`.  We shouldn't wait up for that to be implemented so we'll just move forward with mock for now..

https://github.com/pallets/click/issues/1720

Reply:
Mocking is a very powerful tool that allows you to isolate the code components you want to test and assume that the dependant component (the one you are mocking) is working correctly. Using mocks can make your tests clearer, simpler, single-responsibility driven, and shorter.

I understand why people don't like mocks. When people do not use them correctly, they can cause a lot of problems. But, if you do know how to use them and keep using them **the right way**, it will change the way you test your code for life!

Reply:
Is that better than the library you would otherwise mock providing an explicitly documented and supported feature such that you don't have to mock?

Reply:
Actually, in my opinion: no. I have 3 reasons for that:

1. *python* mocking library is based on **mock injection**. That means that when you use the python mock library you do not need to alter your code in order for it to be mockable (`mock.patch` is a good example for that).
2. Libraries should not "invent the wheel" and create their own mocking systems. Very smart people at the PSF have thought this through and created a general mocking library that can satisfy most needs.
3. When you rely heavily on a specific dependant library in your testing (in our case, the *click* library) you make it harder for future developers to replace this library if needed. One day you might want to use a different library for arguments parsing (maybe use the good old `argparse` 😉 ), and it will be very annoying to remove all the tests you've already written just because you decided to replace implementation. a good guideline for testing is **Test Behaviors, Not Implementations**. When you use the library's mocking process, you test implementation, not behavior.

As for everything in code, there is no "right" and "wrong", but it's a matter of approach. I hope that the reasons I mentioned above are well reasoned, but I'm sure that different people have different opinions about it.